interfaces/builtin/custom_device.go

```diff
@@ -451,8 +434,43 @@ func (iface *customDeviceInterface) UDevConnectedPlug(spec *udev.Specification,
 	}
 
 	// Now write all the rules
-	for _, rule := range deviceRules {
-		spec.TagDevice(rule)
+	for deviceName, rule := range deviceRules {
+		if rule != placeholderRule {
+			spec.TagDevice(rule)
+			continue
+		}
+
+		baseName := filepath.Base(deviceName)
+
+		defaultRule := fmt.Sprintf(`KERNEL=="%s"`, deviceName)
+		defaultBaseNameRule := fmt.Sprintf(`KERNEL=="%s"`, baseName)
+
+		if baseName == deviceName {
+			spec.TagDevice(defaultRule)
+			continue
+		}
+
+		baseNameRule, exists := deviceRules[baseName]
+		if !exists {
+			// There is no rule for the basename, so emit a default
+			// rule for both the full path and basename.
+			spec.TagDevice(defaultRule)
+			spec.TagDevice(defaultBaseNameRule)
+			continue
+		}
+
+		if baseNameRule != placeholderRule && !strutil.ListContains(allDevicePaths, "/dev/"+baseName) {
+			// There is a user-defined rule for the basename of the
+			// device path, and there is not a device whose path
+			// is /dev/<basename>, so that rule should apply to
+			// the device given by this full path. Thus, do not
+			// emit a default rule for this device name.
+			// validateUDevTaggingRule() already checked that the
+			// basename rule only applies to only one device.
```

Comment:
I still think it might be worth logging this situation as it might help debug in cases that the full path was needed but the basename was used

Reply:
Do you think `logger.Noticef()` or `logger.Debugf()` is appropriate here? Or something else?

Reply:
Noticef